Select linters & project logo.

Prikalel opened the pull request:
Solve https://github.com/KH9IZ/task_tree_local/issues/7
Add steps to run linters on your repo local copy.

Commit:
Translate ru section to english.

* Solve TODO. Add link to the issue of 'Task-Reaction system'.

(cherry picked from commit 80d04f79942bab9822fbcd0a5ac574cfffceaa11)

Commit:
Add github ci settings and linters.

Commit:
Add docstrings so linters will not throw errors.

Commit:
Fix spelling mistake in a file name.

KH9IZ commented on .github/workflows/push_to_master.yml:
```diff
@@ -0,0 +1,26 @@
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
```
It think we should use last 3.10 version of python

Prikalel replied:
Ahah, we actually should use 3.6, you know...![IMG_20220330_121717.jpg](https://user-images.githubusercontent.com/67920266/160797212-f41f9d5b-2661-4682-9860-8b960863cc72.jpg)

Prikalel replied:
> It think we should use last 3.10 version of python

Last version is cool, but there are people, who do not follow last news. So, they have, for example 3.6 version, but some new features in 3.10 will not work in 3.6. That is why these people will not upgrade the python on their machines just for one package. I'm sure we should use not laster version than 3.8.

gen-gematogen replied:
It seams strange for me to use older python, because most servers use 3.6. Because project development on older python partially forces users not to update they're python. Also many client machines run 3.8 now, because recent distros support it.

KH9IZ commented:
There is no need to breed entities. I thought that would be one - two config files with configs for pylint and pydocstyle.
Not directory with about 10 linters and separate shell script and even README for that.

Github actions is pretty good, by the way!

Commit:
Remove 10 linters. Add just 2. Patch pylama.

Prikalel commented:
> There is no need to breed entities. I thought that would be one - two config files with configs for pylint and pydocstyle. Not directory with about 10 linters and separate shell script and even README for that.
> 
> Github actions is pretty good, by the way!

Please, review the request.
I have deleted 10 linters and add just 2.
`Black` for autoformatting and `pylama` which is (suddenly!) the wrapper around all other that I used before.
we have the issue for pylama (see https://github.com/klen/pylama/issues/214 ) so I added the patch that will fix the problem.

Also fix github action so before using pylama it will patch the package.

Prikalel commented:
I will add documentation to the README about our linters, so, please, review it and do not merge, until I fix README.

Commit:
Remove ci settings.

Commit:
Remove pylama files (config  and patch).

Commit:
Update project logo.

Commit:
Add linters and code checkers info to documentation.

Prikalel commented:
Ok, we decided to use `black` and `pylint` and do not use CI.
Also I changed the logo as mentioned here https://github.com/Chauss-LLC/task_tree_local/issues/3.

Commit:
Fix linters to be `pylint` and `pydocstyle`.

Commit:
Add pylint configuration file.

Prikalel commented:
Ok, we finally decided to use `pydocstyle` & `pylint`.
See [configuration file](https://github.com/Prikalel/task_tree_local/blob/bc8fb2fc287ad11bb387833dd75ffc0bed5a530b/.pylintrc) for pylint in the root directory.

Prikalel commented:
Now to run checkers you should use:
```
python -m pylint --rcfile=.pylintrc --py-version=3.8 $(git ls-files '*.py')
python -m pydocstyle --convention=google $(git ls-files '*.py')
```
Currently now they do not throw any erros.

P.S. Of cource, I added these instructions to the README.

Commit:
Fix arguments in `pydocstyle` checker call.

* We will save our tests in `tests` folder.
